storage/keyring.go

```diff
@@ -165,34 +168,124 @@ var storageDataUnmarshal Unmarshaler = json.Unmarshal
 
 // loads the entire StorageData into memory
 func (kr *KeyringStore) getStorageData(s *StorageData) error {
-	data, err := kr.keyring.Get(RECORD_KEY)
+	var data []byte
+	var err error
+
+	switch kr.getMyGOOS() {
+	case "windows":
+		data, err = kr.joinAndGetKeyringData(RECORD_KEY)
+	default:
+		data, err = kr.getKeyringData(RECORD_KEY)
+	}
+
 	if err != nil {
 		// Didn't find anything in our keyring
 		*s = NewStorageData()
 		return nil
 	}
-	if err = storageDataUnmarshal(data.Data, s); err != nil {
+	if err = storageDataUnmarshal(data, s); err != nil {
 		return err
 	}
 	return nil
 }
 
+// reads a single entry of the keyring
+func (kr *KeyringStore) getKeyringData(key string) ([]byte, error) {
+	data, err := kr.keyring.Get(key)
+	if err != nil {
+		return nil, err
+	}
+	return data.Data, nil
+}
+
+// read the chunks stored in windows credential manager
+func (kr *KeyringStore) joinAndGetKeyringData(key string) ([]byte, error) {
+	i := 0
+	data := []byte{}
+	var err error
+	var payload_size uint64
+	var chunk []byte
+
+	chunk, err = kr.getKeyringData(fmt.Sprintf("%s_%d", key, i))
+	if err != nil {
+		return nil, err
+	}
+	if len(chunk) < 8 {
+		return nil, fmt.Errorf("Invalid stored data in wincred. There's not enough data")
```

Comment:
Codecov is still saying there is still a missing unit test for the first read being < 8 bytes.

also, I should mention the error shouldn't mention `wincred` since we're not actually sure they are using that keystore, since this code happens anytime the user is running on Windows.

Reply:
Solved :)